core/objects.go

```diff
@@ -254,6 +274,10 @@ type Challenge struct {
 
 	// Used by dns and dvsni challenges
 	Validation *jose.JsonWebSignature `json:"validation,omitempty"`
+
+	// Contains information about URLs used or redirected to and IPs resolved and
+	// used
+	ValidationRecord *ValidationRecord `json:"validationRecord,omitempty"`
```

Comment:
I honestly don't care whether an array makes sense for other validation types.  A one-entry array is not that much more expensive than the thing itself.

Architectural consistency is more important than future-proofing.  For better or worse, the architecture we have now says that the "type" element of the challenge dictates the structure of the rest of the challenge.  So we have to have "type"-specific logic no matter what.  This sub-object stuff adds no value.